scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -91,9 +91,9 @@ def load_modules(self, modules_to_load):
         if (module_system == "module"):
             self._load_module_modules(modules_to_load)
         elif (module_system == "soft"):
-            self._load_soft_modules(modules_to_load)
-        elif (module_system == "dotkit"):
-            self._load_dotkit_modules(modules_to_load)
+            self._load_modules_generic(modules_to_load)
```

Comment:
Encapsulate environment systems that do not have a python interface into "generic".